### PR TITLE
[FIRRTL] Query for target result instead of rejecting with multiple results.

### PR DESCRIPTION
**Pull request by dtzSiFive**
Fix annotation resolution, GC view targeting, and resolve traces usage that target forceable declarations.

Builds on #4966 .

### CODE REVIEW

File: lib/Dialect/FIRRTL/Transforms/ResolveTraces.cpp [hunk 1]
```diff
@@ -148,14 +148,21 @@ struct ResolveTracesPass : public ResolveTracesBase<ResolveTracesPass> {
 
     // If this is operation doesn't have a single result (and no way to know
     // what its type is) or if it doesn't have a name, then do nothing.
```
Comment (mikeurbach) [inline lib/Dialect/FIRRTL/Transforms/ResolveTraces.cpp:150]: Maybe this comment should move down with the `getNumResults()` check?

**Review by mikeurbach — COMMENTED** [review]
I'm not super familiar with this, but looks good to me. Hooray for interfaces.

File: lib/Dialect/FIRRTL/Transforms/ResolveTraces.cpp (hunk 1 above)
Comment (dtzSiFive) [reply lib/Dialect/FIRRTL/Transforms/ResolveTraces.cpp]: Updated this comment block, thanks!

**Comment by dtzSiFive** [comment]
Thanks!

**Pull request merged**